### PR TITLE
RSDK-1878 - add mpu6050 tests for readings

### PR DESCRIPTION
**Pull request by gvaradarajan**

### CODE REVIEW

File: components/movementsensor/mpu6050/mpu6050_test.go [hunk 1]
```diff
@@ -178,3 +179,129 @@ func TestSuccessfulInitializationAndClose(t *testing.T) {
...
+	linearAccelMockData[0] = 64
+	linearAccelMockData[1] = 0
+	expectedAccelX := 9810.0
+	// y-accel
+	linearAccelMockData[2] = 32
+	linearAccelMockData[3] = 0
+	expectedAccelY := 4905.0
+	// z-accel
+	linearAccelMockData[4] = 16
+	linearAccelMockData[5] = 0
```
Comment (kim-mishra) [inline components/movementsensor/mpu6050/mpu6050_test.go:235]: Why are there 2 values for each dimension?

File: components/movementsensor/mpu6050/mpu6050_test.go [hunk 2]
```diff
@@ -178,3 +179,129 @@ func TestSuccessfulInitializationAndClose(t *testing.T) {
...
+
+	logger := golog.NewTestLogger(t)
+	cfg, deps := setupDependencies(angVelMockData)
+	sensor, err := NewMpu6050(context.Background(), deps, cfg, logger)
+	test.That(t, err, test.ShouldBeNil)
+	defer utils.TryClose(context.Background(), sensor)
+	testutils.WaitForAssertion(t, func(tb testing.TB) {
+		angVel, err := sensor.AngularVelocity(context.Background(), nil)
+		test.That(tb, err, test.ShouldBeNil)
+		test.That(tb, angVel, test.ShouldNotBeZeroValue)
```
Comment (kim-mishra) [inline components/movementsensor/mpu6050/mpu6050_test.go:279]: Why use `test.ShouldNotBeZeroValue` here and not check that it is or is close to the mock values you set above?

File: components/movementsensor/mpu6050/mpu6050_test.go (hunk 1 above)
Comment (gvaradarajan) [reply components/movementsensor/mpu6050/mpu6050_test.go:235]: each dimension comes through as 16-bit BigEndian values

File: components/movementsensor/mpu6050/mpu6050_test.go (hunk 2 above)
Comment (gvaradarajan) [reply components/movementsensor/mpu6050/mpu6050_test.go:279]: this is a wait function to make sure that we don't test the value until the coroutine that updates the struct has run at least once. It'll block the test function until the angular velocity is a non-zero value, then check the accuracy of the actual angular velocity value below

File: components/movementsensor/mpu6050/mpu6050_test.go [hunk 3]
```diff
@@ -178,3 +179,129 @@ func TestSuccessfulInitializationAndClose(t *testing.T) {
...
+		return i2c, true
+	}
+	return cfg, registry.Dependencies{
+		resource.NameFromSubtype(board.Subtype, testBoardName): mockBoard,
+	}
+}
+
+//nolint:dupl
+func TestLinearAcceleration(t *testing.T) {
+	linearAccelMockData := make([]byte, 16)
```
Comment (penguinland) [inline components/movementsensor/mpu6050/mpu6050_test.go:224]: Put in a comment reminding the reader that, although this function only tests values from the first 6 bytes of data, the driver reads 16 bytes from the sensor, so we need to provide 16 bytes in the mock.

**Review by penguinland — APPROVED** [review]
LGTM! Feel free to skip my comment if you don't think it's important.

File: components/movementsensor/mpu6050/mpu6050_test.go (hunk 2 above)
Comment (penguinland) [reply components/movementsensor/mpu6050/mpu6050_test.go:279]: Consider putting that response in a comment in the code. If Kim didn't see that, someone else is going to miss it, too.

**Pull request merged**